Index: utils.c
===================================================================
--- utils.c	(revision 890)
+++ utils.c	(working copy)
@@ -56,7 +56,7 @@
   int 			len;
   char 			*res = NULL;
 
-  if ((len = getNumber(p, l)) > 0) {
+  if ((len = getNumber(p, l)) > 0 && len < NOTANUMBER) {
     res = malloc(len + 1);
     res[len] = '\0';
     strncpy(res, *p, len);
Index: main.c
===================================================================
--- main.c	(revision 860)
+++ main.c	(working copy)
@@ -58,6 +58,8 @@
   int			isAdmin = 0;
   int			i;

+  if (!password)
+    return NOBODY;
   if (!strcmp(password, userPassword))
     isUser = 1;
   strcpy(savePassword, password);
